community/kernel/src/test/java/org/neo4j/kernel/impl/transaction/log/checkpoint/CheckPointerIntegrationTest.java

```diff
@@ -0,0 +1,224 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.transaction.log.checkpoint;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.graphdb.factory.GraphDatabaseBuilder;
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.kernel.impl.transaction.log.LogEntryCursor;
+import org.neo4j.kernel.impl.transaction.log.LogVersionedStoreChannel;
+import org.neo4j.kernel.impl.transaction.log.PhysicalLogFile;
+import org.neo4j.kernel.impl.transaction.log.PhysicalLogFiles;
+import org.neo4j.kernel.impl.transaction.log.ReadAheadLogChannel;
+import org.neo4j.kernel.impl.transaction.log.ReadableVersionableLogChannel;
+import org.neo4j.kernel.impl.transaction.log.entry.CheckPoint;
+import org.neo4j.kernel.impl.transaction.log.entry.LogEntry;
+import org.neo4j.kernel.impl.transaction.log.entry.LogEntryReader;
+import org.neo4j.kernel.impl.transaction.log.entry.LogEntryReaderFactory;
+import org.neo4j.test.EphemeralFileSystemRule;
+import org.neo4j.test.TestGraphDatabaseFactory;
+
+import static org.junit.Assert.assertEquals;
+import static org.neo4j.kernel.impl.transaction.log.LogVersionBridge.NO_MORE_CHANNELS;
+import static org.neo4j.kernel.impl.transaction.log.LogVersionRepository.INITIAL_LOG_VERSION;
+import static org.neo4j.kernel.impl.transaction.log.ReadAheadLogChannel.DEFAULT_READ_AHEAD_SIZE;
+
+public class CheckPointerIntegrationTest
+{
+    private static final File storeDir = new File( "/tmp/graph.db" );
+
+    @Rule
+    public EphemeralFileSystemRule fsRule = new EphemeralFileSystemRule();
+
+    private GraphDatabaseBuilder builder;
+    private FileSystemAbstraction fs;
+
+    @Before
+    public void setup() throws IOException
+    {
+        fs = fsRule.get();
+        fs.deleteRecursively( storeDir );
+        builder = new TestGraphDatabaseFactory().setFileSystem( fs ).newImpermanentDatabaseBuilder( storeDir );
+    }
+
+    @Test
+    public void shouldCheckPointBasedOnTime() throws Throwable
+    {
+        // given
+        long seconds = 3;
+        GraphDatabaseService db = builder
+                .setConfig( GraphDatabaseSettings.check_point_interval_time, seconds + "s" )
+                .setConfig( GraphDatabaseSettings.check_point_interval_tx, "10000" )
+                .setConfig( GraphDatabaseSettings.logical_log_rotation_threshold, "1g" )
+                .newGraphDatabase();
+
+        // when
+        try ( Transaction tx = db.beginTx() )
+        {
+            db.createNode();
+            tx.success();
+        }
+
+        // trigger the check point
+        Thread.sleep( TimeUnit.SECONDS.toMillis( seconds ) );
```

Comment:
I would have gone with a tracer but they are not easily configurable and definitely not via graph database configuration. I am unsure if the extra complexity will pay off, since this test never failed any PR build so far.